addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiAssociationCommandClass.java

```diff
@@ -252,22 +254,22 @@ protected void processGroupingsReport(SerialMessage serialMessage, int offset) t
      * @return the serial message
      */
     public SerialMessage setAssociationMessage(int group, int node, int endpoint) {
-        logger.debug("NODE {}: Creating new message for command MULTI_ASSOCIATIONCMD_SET", getNode().getNodeId());
+        logger.debug(
+                "NODE {}: Creating new message for command MULTI_ASSOCIATIONCMD_SET node {}, endpoint {}, group {}",
+                getNode().getNodeId(), node, endpoint, group);
         SerialMessage result = new SerialMessage(getNode().getNodeId(), SerialMessageClass.SendData,
                 SerialMessageType.Request, SerialMessageClass.SendData, SerialMessagePriority.Set);
 
-        if (endpoint == 0) {
-            logger.trace("NODE {}: Endpoint is 0. Sending only node.", getNode().getNodeId());
-            byte[] newPayload = { (byte) getNode().getNodeId(), 4, (byte) getCommandClass().getKey(),
-                    (byte) MULTI_ASSOCIATIONCMD_SET, (byte) (group & 0xff), (byte) (node & 0xff) };
-            result.setMessagePayload(newPayload);
-        } else {
```

Comment:
The old code made a distinction between a association to a node (ie endpont = 0 ) and an association node with an endpoint.  In my opinion is safer not to assume the nodes can handle the  case where you create an association with endpoint where the endpoint is 0.